tests/components/nest/sensor_sdm_test.py

```diff
@@ -0,0 +1,210 @@
+"""
+Test for Nest sensors platform for the Smart Device Management API.
+
+These tests fake out the subscriber/devicemanager, and are not using a real
+pubsub subscriber.
+"""
+
+from google_nest_sdm.device import Device
+from google_nest_sdm.event import EventMessage
+
+from .common import async_setup_sdm_platform
+
+PLATFORM = "sensor"
+
+THERMOSTAT_TYPE = "sdm.devices.types.THERMOSTAT"
+
+
+async def async_setup_sensor(hass, devices={}, structures={}):
+    """Set up the platform and prerequisites."""
+    return await async_setup_sdm_platform(hass, PLATFORM, devices, structures)
+
+
+async def test_thermostat_device(hass):
+    """Test a thermostat with temperature and humidity sensors."""
+    devices = {
+        "some-device-id": Device.MakeDevice(
+            {
+                "name": "some-device-id",
+                "type": THERMOSTAT_TYPE,
+                "traits": {
+                    "sdm.devices.traits.Info": {
+                        "customName": "My Sensor",
+                    },
+                    "sdm.devices.traits.Temperature": {
+                        "ambientTemperatureCelsius": 25.1,
+                    },
+                    "sdm.devices.traits.Humidity": {
+                        "ambientHumidityPercent": 35.0,
+                    },
+                },
+            },
+            auth=None,
+        )
+    }
+    await async_setup_sensor(hass, devices)
+
+    temperature = hass.states.get("sensor.my_sensor_temperature")
+    assert temperature is not None
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)